File: sdks/sdkwork-mail-sdk/sdkwork-mail-sdk-go/Mailstandard/provider_activation_catalog.go
package Mailstandard

type MailProviderActivationCatalogEntry struct {
    ProviderKey      string
    PluginId         string
    DriverId         string
    ActivationStatus string
    RuntimeBridge    bool
    RootPublic       bool
    PackageBoundary  bool
    Builtin          bool
    PackageIdentity  string
}

type MailProviderActivationCatalog struct{}

var mail_PROVIDER_ACTIVATION_STATUSES = []string{"package-boundary", "control-metadata-only"}

var OFFICIAL_mail_PROVIDER_ACTIVATIONS = []MailProviderActivationCatalogEntry{
    {ProviderKey: "smtp", PluginId: "Mail-smtp", DriverId: "sdkwork-mail-driver-smtp", ActivationStatus: "package-boundary", RuntimeBridge: true, RootPublic: false, PackageBoundary: true, Builtin: true, PackageIdentity: "github.com/sdkwork/Mail-sdk-provider-smtp"},
    {ProviderKey: "imap", PluginId: "Mail-imap", DriverId: "sdkwork-mail-driver-imap", ActivationStatus: "package-boundary", RuntimeBridge: true, RootPublic: false, PackageBoundary: true, Builtin: true, PackageIdentity: "github.com/sdkwork/Mail-sdk-provider-imap"},
}

func GetMailProviderActivationByProviderKey(providerKey string) *MailProviderActivationCatalogEntry {
    for index := range OFFICIAL_mail_PROVIDER_ACTIVATIONS {
        if OFFICIAL_mail_PROVIDER_ACTIVATIONS[index].ProviderKey == providerKey {
            return &OFFICIAL_mail_PROVIDER_ACTIVATIONS[index]
        }
    }

    return nil
}

File: sdks/sdkwork-mail-sdk/sdkwork-mail-sdk-rust/src/provider_activation_catalog.rs
#[derive(Clone, Copy)]
#[allow(non_snake_case)]
pub struct MailProviderActivationCatalogEntry {
    pub providerKey: &'static str,
    pub pluginId: &'static str,
    pub driverId: &'static str,
    pub activationStatus: &'static str,
    pub runtimeBridge: bool,
    pub rootPublic: bool,
    pub packageBoundary: bool,
    pub builtin: bool,
    pub packageIdentity: &'static str,
}

pub struct MailProviderActivationCatalog;

pub const mail_PROVIDER_ACTIVATION_STATUSES: [&str; 2] = ["package-boundary", "control-metadata-only"];

pub const OFFICIAL_mail_PROVIDER_ACTIVATIONS: [MailProviderActivationCatalogEntry; 2] = [
    MailProviderActivationCatalogEntry { providerKey: "smtp", pluginId: "Mail-smtp", driverId: "sdkwork-mail-driver-smtp", activationStatus: "package-boundary", runtimeBridge: true, rootPublic: false, packageBoundary: true, builtin: true, packageIdentity: "Mail-sdk-provider-smtp" },
    MailProviderActivationCatalogEntry { providerKey: "imap", pluginId: "Mail-imap", driverId: "sdkwork-mail-driver-imap", activationStatus: "package-boundary", runtimeBridge: true, rootPublic: false, packageBoundary: true, builtin: true, packageIdentity: "Mail-sdk-provider-imap" },
];

pub fn get_mail_provider_activation_by_provider_key(
    provider_key: &str,
) -> Option<&'static MailProviderActivationCatalogEntry> {
    OFFICIAL_mail_PROVIDER_ACTIVATIONS
        .iter()
        .find(|entry| entry.providerKey == provider_key)
}
